whois-api/src/main/java/net/ripe/db/whois/api/fulltextsearch/ElasticFulltextSearch.java

```diff
@@ -112,50 +115,69 @@ protected SearchResponse doSearch() throws IOException {
                 final SearchHit[] hits = fulltextResponse.getHits().getHits();
 
                 LOGGER.debug("ElasticSearch {} hits for the query: {}", hits.length, searchRequest.getQuery());
-                final List<RpslObject> results = Lists.newArrayList();
-                int resultSize = Math.min(maxResultSize,Long.valueOf(fulltextResponse.getHits().getTotalHits().value).intValue());
+                return prepareResponse(fulltextResponse, hits, searchRequest, stopwatch);
+            }
+        }.search();
+    }
+
+    private SearchResponse prepareResponse(final org.elasticsearch.action.search.SearchResponse fulltextResponse,
+                                           final SearchHit[] hits,
+                                           final SearchRequest searchRequest,
+                                           final Stopwatch stopwatch) {
+        final List<SearchResponse.Lst> highlightDocs = Lists.newArrayList();
+        final List<SearchResponse.Result.Doc> resultDocumentList = Lists.newArrayList();
+        fillHighlightsAndResults(hits, highlightDocs, resultDocumentList);
 
```

Comment:
Can you separate the highlights and results? Anyways you are calling  createHighlights inside the function

Reply:
I spent some time thinking on this, if I do that I need to perform another loop, exactly the same look for hits.
Finally I choose to do everything in the same loop and putting a good name to the method. I could split in two but I am not fancy of that idea

Reply:
we can discuss abut it. 
For getting the search response we need to loop hits
For getting the highlight response the same....we need to loop hits

I don't want to loop hits twice